app/src/main/java/com/example/fitjournal/components/dialogs/FilterJournalDialog.kt

```diff
@@ -0,0 +1,256 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.window.DialogProperties
+import com.example.fitjournal.R
+import com.example.fitjournal.model.enum.WorkoutTypeEnum
+import com.example.fitjournal.model.events.HomeScreenEvents
+import com.example.fitjournal.model.ui.FilterWorkoutModel
+import com.example.fitjournal.theme.Spacing
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun FilterJournalDialog(
+    modifier: Modifier = Modifier,
+    properties: DialogProperties = DialogProperties(),
+    workoutList: List<FilterWorkoutModel>,
+    homeScreenEvents: (HomeScreenEvents) -> Unit
+) {
+    BasicAlertDialog(
+        onDismissRequest = {
+            homeScreenEvents(HomeScreenEvents.DismissFilterExercisesDialog)
+        },
+        modifier = modifier,
+        properties = properties,
+        content = {
+            FilterJournalSection(
+                modifier = modifier,
+                workoutList = workoutList,
+                onDismissDialog = {
+                    homeScreenEvents(HomeScreenEvents.DismissFilterExercisesDialog)
+                },
+                onConfirmDialog = {
+                    homeScreenEvents(HomeScreenEvents.OnConfirmFilterExercisesDialog(it))
+                }
+            )
+        }
+    )
+}
+
+@Composable
+fun FilterJournalSection(
+    modifier: Modifier = Modifier,
+    workoutList: List<FilterWorkoutModel>,
+    onDismissDialog: () -> Unit,
+    onConfirmDialog: (List<WorkoutTypeEnum>) -> Unit,
+) {
+    val mutableWorkoutList: MutableList<WorkoutTypeEnum> = workoutList.mapNotNull {
+        if (it.isWorkoutSelected) it.exerciseType
+        else null
+    }.toMutableList()
+    Column(
+        modifier = modifier,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        FilterJournalDialogTitle()
+        Spacer(modifier = Modifier.height(Spacing.spacing12))
+        FilterJournalDialogSubtitle()
+        Spacer(modifier = Modifier.height(Spacing.spacing12))
+        FilterJournalDialogCheckboxes(
+            workoutList = workoutList,
+            onFilterCheckboxChanged = { workoutTypeEnum, checkboxValue ->
+                if (checkboxValue){
+                    if (!mutableWorkoutList.contains(workoutTypeEnum)) {
+                        mutableWorkoutList.add(workoutTypeEnum)
+                    }
+                }
+                if (!checkboxValue){
+                    if (mutableWorkoutList.contains(workoutTypeEnum)) {
+                        mutableWorkoutList.remove(workoutTypeEnum)
+                    }
+                }
```

Comment:
We can combine these if statements into:
```suggestion
                if (checkboxValue){
                    if (!mutableWorkoutList.contains(workoutTypeEnum)) {
                        mutableWorkoutList.add(workoutTypeEnum)
                    }
                } else {
                    if (mutableWorkoutList.contains(workoutTypeEnum)) {
                        mutableWorkoutList.remove(workoutTypeEnum)
                    }
                }
```

However, having nested if statements is considered bad practice so maybe we can do:
```suggestion
               when (checkboxValue) {
                   true -> {
                          if (!mutableWorkoutList.contains(workoutTypeEnum)) {
                                 mutableWorkoutList.add(workoutTypeEnum)
                          }
                    }
                   false -> {
                          if (mutableWorkoutList.contains(workoutTypeEnum)) {
                                 mutableWorkoutList.remove(workoutTypeEnum)
                          }
                   }
              }
```

Reply:
I dont think the nesting here creates any more complexity as the if statements themselves are very readable and self explanitory